packages/core/src/common/errors.ts

```diff
@@ -120,3 +120,16 @@ export const OVERLAY_WITH_MULTIPLE_CHILDREN_REQUIRES_CHILD_REFS =
     ns + ` <Overlay2> requires childRefs prop when rendering multiple child elements`;
 export const OVERLAY_CHILD_REQUIRES_KEY =
     ns + ` <Overlay2> requires each child element to have a unique key prop when childRefs is used`;
+
+function deprecatedSizeWarning(component: string, size: "large" | "small") {
```

Comment:
Thoughts on just using this function directly from components? I feel like we're going to have a lot of constants resulting from this that will all just be used in one place, might be nicer to just use this function in `useValidateProps`.

Reply:
If we wanted to get extra fancy we could do something like:


```ts
function logDeprecatedSizeWarning(
    component: string,
    props: Partial<Record<"large" | "small", boolean>>,
) { ... }
```

which would automatically log the right messages based on the passed props which are defined. So in components we could do something like:

```ts
    useValidateProps(() => {
        logDeprecatedSizeWarning("Button", { large, small });
    }, [large, small]);
```

Just a thought, might be too much for a simple case like this.

Reply:
Fair point, I like it! Added in 098390192fc65445791daed58bc49d862142a4aa